### PR TITLE
[F8N-865] Backport globalstats improvements from flagon

### PR DESCRIPTION
**Pull request by colinking**
F8N-865

This PR backports a number of improvements made to `globalstats`, learned from adding support for globalstats-by-default in flagon. Specifically:
- Resolves a bug where `Initialize` could not be called more than once, and where calling more than once would not apply a default 
- In Go 1.12+, the version field will now be populated using Go's debug APIs. Previously, we couldn't report version information when imported as a client.

---

Builds blocked by: https://github.com/segmentio/ctlstore/pull/30

### CODE REVIEW

**Comment by colinking** [comment]
The build step in CI is not running, so this PR is blocked on: https://github.com/segmentio/ctlstore/pull/30

File: pkg/globalstats/stats.go [hunk 1]
```diff
@@ -1,168 +1,248 @@
...
-		}
-
-		return nil
-	}()
-	if err != nil {
-		events.Log("Could not initialize ctlstore globalstats: %{error}s", err)
-		return nil, false
+	// Validate that all config values are valid.
+	if cfg.FlushEvery < 0 {
+		events.Log("Could not initialize flagon global stats: %{error}s", errors.New("flush rate must be a positive duration"))
```
Comment (collinvandyck) [inline pkg/globalstats/stats.go]: s/flagon/ctlstore

Comment (colinking) [reply pkg/globalstats/stats.go]: Good catch! Thought I grepped for `flagon` before pushing, but I guess not 😂

**Comment by collinvandyck** [comment]
small nit: rebasing and force pushing breaks links in github emails. i usually just add additional commits after feedback is given to preserve history.

**Comment by collinvandyck** [comment]
Not a big deal but it would be easier just to review the one small change you made instead of needing to look at the entire diff again to see what you've changed.

**Comment by colinking** [comment]
Oof, good point! I can do merges in the future instead of rebasing changes + applying changes as follow-on commits if that renders better :)

**Pull request merged**